mod/blog/views/default/widgets/blog/content.php

```diff
@@ -3,29 +3,27 @@
  * User blog widget display view
  */
 
-$num = $vars['entity']->num_display;
+$widget = elgg_extract('entity', $vars);
 
-$options = array(
+$content = elgg_list_entities([
 	'type' => 'object',
 	'subtype' => 'blog',
-	'container_guid' => $vars['entity']->owner_guid,
-	'limit' => $num,
-	'full_view' => false,
+	'container_guid' => $widget->owner_guid,
+	'limit' => $widget->num_display,
 	'pagination' => false,
 	'distinct' => false,
-);
-$content = elgg_list_entities($options);
+]);
 
-echo $content;
-
-if ($content) {
-	$blog_url = "blog/owner/" . elgg_get_page_owner_entity()->username;
-	$more_link = elgg_view('output/url', array(
-		'href' => $blog_url,
-		'text' => elgg_echo('blog:moreblogs'),
-		'is_trusted' => true,
-	));
-	echo "<span class=\"elgg-widget-more\">$more_link</span>";
-} else {
+if (empty($content)) {
 	echo elgg_echo('blog:noblogs');
+	return;
 }
+
+echo $content;
+
+$more_link = elgg_view('output/url', [
+	'href' => 'blog/owner/' . $widget->getOwnerEntity()->username,
+	'text' => elgg_echo('blog:moreblogs'),
+	'is_trusted' => true,
+]);
+echo "<div class=\"elgg-widget-more\">$more_link</div>";
```

Comment:
Should we just add `more_link` list options and teach list component to use it instead of pagination?